src/report_tests.py

```diff
@@ -14,7 +14,9 @@ def __init__(self, artifact_path: str) -> None:
         self.text = ''
 
     def append(self, status: str, test: str, comment: str = '') -> None:
-        link = Markdown.link(test, self.artifact_path + test)
+        name = test[test.rfind('/') + 1:test.rfind('.log')]
+        name = name.replace('_', ' ')
```

Comment:
I don't think it's possible to recover the test name from the sanitized version. `zlib_ng` prevents this and also the `%` in front of the compiler gets lost. Plus IMHO `spack install eccodes %nvhpc` and `spack install eccodes %pgi` should not be listed as two different tests, because they try to test the same thing, that eccodes can be build with the fancy compiler from nvidia. They are in the same test function because of that reason. Maybe the actual function name of the test could serve as a label here. This is possible to do in Python; I think the master does it. That name could be written into the log file on a single line and then be recovered.

Reply:
Or the name of the log file could be changed to it, that's maybe even simpler.

Reply:
The name of the logfiles is now the class name + function name.